### ISSUE TITLE
MS Office "Share as link" feature not working properly

### ISSUE THREAD

**Issue by benjlevesque** [WebDAVSharp/WebDAVSharp.Server#13]
Hello, 
I can't get to work the feature that allows to share a doc by email :

Sharepoint has a similar feature, and the link is formated as https://sharepoint.mydomain.com. 
With WebDAVSharp.Server, I expect https://webdav.mydomain.com but I get 
file:///\webdav.mydomain.com@SSL\DavWWWRoot\

this is a big problem because it implies the users to all have the same configuration on their PC.

Any workaround?

Thanks

**Comment by Winterleaf** [comment]
Morning, I am currently using the webdav as a file server.  We mapped the \\<server>\DavWWWRoot\<User Folder> to there network drives when they log in.  After doing this we had no problems with users using the server as a network share.  You have to be careful, windows will try and change the unc path to \\server\path  and drop the DavWWWRoot.

**Comment by Winterleaf** [comment]
Also, if you could post screenshot with what your talking about circled it would help me trace down the issue faster.  I have an idea that might fix it.

**Comment by EMerckx** [comment]
@benjlevesque and I communicated about this a while back, and he told me that adding a piece of SharePoint's header to the OPTIONS method worked.

In OPTIONS, adding the following code did the trick for him.

``` C#
context.Response.AppendHeader("X-MSDAVEXT", "1");
context.Response.AppendHeader("DocumentManagementServer", "Properties Schema;");
context.Response.AppendHeader("X-MSFSSHTTP", "1.2");
var requestUri = GetRequestUri(context.Request.Url.ToString());
context.Response.AppendHeader("X-MSWEBURL", requestUri.AbsoluteUri);
context.Response.AppendHeader("DAV", "1,2");
context.Response.AppendHeader("Public-Extension", "http://schemas.microsoft.com/repl-2");
```

So he suggested to include an extension to the server, which is not in the RFC, in order to support Office.

**Comment by mikeobrien** [comment]
I spent a bunch of time on getting this working with Excel (migrating off of MOSS 2007). Did a ton of experimentation, even added some Front Page Extensions functionality. In the end, once I boiled it all down, the only thing I needed to add was this `OPTIONS` header:

```csharp
context.Response.AppendHeader("MS-Author-Via", "MS-FP/4.0,DAV");
```
Without this header Excel would open it in readonly mode and not let me save it the WebDAV server.

This is the url we used:

```
ms-excel:ofv|u|http://mywebdav.org/SomeFile.xlsx
```
